### DISCUSSION TITLE
Seeing a lot of waiting time between tasks

### DISCUSSION THREAD

**Discussion by skreddy6673** [concourse/concourse#6631] [General]
![image](https://user-images.githubusercontent.com/27447138/109848066-de46a680-7c1d-11eb-8751-9ea8b5dc8695.png)

Seeing a waiting time between tasks in concourse , not sure how to troubleshoot this. Any suggestions would be highly appreciable.  

Helm: 12.0.1
Container Placement Strategy: Random

**Comment by taylorsilva** [comment]
First guess is volume streaming. You can verify this by seeing which workers each step landed on. I can see in the second step it started on `concourse-prod-worker-9`. Which worker did the previous step start on? If it's a different worker (likely given your container placement strategy) then most of that start-up time may be related to volume streaming. Enabling tracing is the only way to know for sure.

Also, since your second step is a `put` step, by default all artifacts in the job are streamed to it. You can avoid this by specifying which artifacts should be streamed to it: https://concourse-ci.org/jobs.html#schema.step.put-step.inputs

**Reply by skreddy6673** [reply]
The previous step executed on worker-16.

**Reply by skreddy6673** [reply]
Specifying which artifacts should be streamed made a difference, will be testing this on bigger pipelines. Thank you @taylorsilva .
